src/screens/groups/components/batch-recipients-list/batch-recipients-list.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { Checkbox, Table } from 'semantic-ui-react'
+import type { RecipientDTO } from '@/lib/dto/RecipientDTO'
+import type { SelectedRecipients } from '@/screens/groups/batch-payment-screen'
+import BatchRecipientInfo from './batch-recipient-info'
+
+interface Props {
+  recipients: RecipientDTO[]
+  selectedRecipients: SelectedRecipients
+  onRowSelect: (_id: string) => void
+  onAmountChange: (_id: string, _amount: number) => void
+  onSelectAll: () => void
+  isSelectAllChecked: boolean
+}
+
+const Component = ({
+  recipients,
+  selectedRecipients,
+  onRowSelect,
+  onAmountChange,
+  onSelectAll,
+  isSelectAllChecked,
+}: Props) => {
+  const handleRowSelect = (id: string) => {
+    const selectedAmount = selectedRecipients[id]?.amount
+    if (selectedAmount === 0) return
```

Comment:
We don't need this extra variable just to check condition. You can simplify:

```
if (parseInt(selectedRecipients[id]?.amount) <= 0) return

onRowSelect(id)
```

Did you check your code with negative numbers? Is that work?

Reply:
yes it works, user can't enter negative values. i have set the minimum value to 0 
![Screenshot from 2023-06-20 13-04-53](https://github.com/optriment/web3-employment/assets/65503300/78a28cf2-51b3-461a-8540-e00b0821a016)

Reply:
upon rechecking still i found a way how user can enter negative value so i have used Math.abs in handleAmountChange so the amount can never be negative 
`const amount = Math.abs(parseFloat(event.target.value)) || 0`
`onAmountChange(id, amount)`